### PR TITLE
Cleanup Raft implementation

### PR DESCRIPTION
**Pull request by eddyashton**
This PR removes some unused features and code from our Raft implementation.

### CODE REVIEW

File: src/consensus/aft/raft.h [hunk 1]
```diff
@@ -1200,7 +908,11 @@ namespace aft
           }
         }
       }
-      else if (consensus_type != ConsensusType::BFT)
+      else if (
+        consensus_type != ConsensusType::BFT &&
+        replica_state != kv::ReplicaState::Retired &&
+        replica_state != kv::ReplicaState::Retiring &&
+        replica_state != kv::ReplicaState::Learner)
```
Comment (jumaffre) [inline src/consensus/aft/raft.h:915]: Is this a functional change for the CFT path?

File: src/node/rpc/frontend.h [hunk 2]
```diff
@@ -547,131 +547,11 @@ namespace ccf
...
-            ctx->set_error(
-              HTTP_STATUS_INTERNAL_SERVER_ERROR,
-              ccf::errors::InternalError,
-              "Consensus is not yet ready.");
-            return ctx->serialise_response();
-          }
-        }
-      }
-
+      // NB: If we want to re-excute on backups, the original command could be
```
Comment (jumaffre) [inline src/node/rpc/frontend.h:550]:
```suggestion
      // NB: If we want to re-execute on backups, the original command could be
```

File: src/consensus/aft/raft.h (hunk 1 above)
Comment (eddyashton) [reply src/consensus/aft/raft.h:915]: Huh, this looks like a bad merge, and is not deliberate. I'll take a look.

Comment (wintersteiger) [reply src/consensus/aft/raft.h]: Those conditions are unnecessary, because `can_endorse_primary` also checks them.

**Review by wintersteiger — APPROVED** [review]

File: src/consensus/aft/raft.h (hunk 1 above)
Comment (eddyashton) [reply src/consensus/aft/raft.h]: Yeah this was a bad merge. Have restored this check to the `else if (consensus_type != ConsensusType::BFT)` that is on `main`.

**Review by achamayou — APPROVED** [review]

**Pull request merged**